domainmessage/msgtx.go

```diff
@@ -302,29 +301,30 @@ func (msg *MsgTx) IsCoinBase() bool {
 // TxHash generates the Hash for the transaction.
 func (msg *MsgTx) TxHash() *daghash.Hash {
 	// Encode the transaction and calculate double sha256 on the result.
-	// Ignore the error returns since the only way the encode could fail
-	// is being out of memory or due to nil pointers, both of which would
-	// cause a run-time panic.
-	buf := bytes.NewBuffer(make([]byte, 0, msg.serializeSize(txEncodingExcludePayload)))
-	_ = msg.serialize(buf, txEncodingExcludePayload)
+	writer := daghash.NewDoubleHashWriter()
+	err := msg.serialize(writer, txEncodingExcludePayload)
+	if err != nil {
+		panic(fmt.Sprintf("This should never fail. err: %+v", err))
```

Comment:
Please be more explicit about what should never happen. 

Give as much info as possible (within the reasonable in an error message) about what happened and why it should never fail.

Reply:
That's a great question :)
I can try, I'll write what I know, but I don't know the serialization code good enough to understand what could fail in it